crates/ordinals/src/brc721/register_collection.rs

```diff
@@ -97,75 +77,42 @@ pub enum RegisterCollectionError {
 	InvalidLength(String),
 }
 
-/// Helper function to ensure the next instruction is a specific opcode.
-///
-/// Returns an error if the expected opcode is not found or if there are no more instructions.
-fn expect_opcode<'a>(
-	instructions: &mut impl Iterator<Item = Result<Instruction<'a>, bitcoin::script::Error>>,
-	expected_op: opcodes::Opcode,
-	desc: &str,
-) -> Result<(), RegisterCollectionError> {
-	match instructions
-		.next()
-		.ok_or_else(|| RegisterCollectionError::InstructionNotFound(desc.into()))?
-	{
-		Ok(Instruction::Op(op)) if op == expected_op => Ok(()),
-		_ => Err(RegisterCollectionError::UnexpectedInstruction),
-	}
-}
-
-/// Helper function to ensure the next instruction is a push operation of the expected length.
-///
-/// Returns an error if the expected length is not met or if there are no more instructions.
-fn expect_push_bytes<'a>(
-	instructions: &mut impl Iterator<Item = Result<Instruction<'a>, bitcoin::script::Error>>,
-	expected_len: usize,
-	desc: &str,
-) -> Result<Vec<u8>, RegisterCollectionError> {
-	match instructions
-		.next()
-		.ok_or_else(|| RegisterCollectionError::InstructionNotFound(desc.into()))?
-	{
-		Ok(Instruction::PushBytes(bytes)) if bytes.len() == expected_len =>
-			Ok(bytes.as_bytes().into()),
-		Ok(Instruction::PushBytes(_)) => Err(RegisterCollectionError::InvalidLength(desc.into())),
-		_ => Err(RegisterCollectionError::UnexpectedInstruction),
-	}
-}
-
 #[cfg(test)]
 mod tests {
 	use super::*;
 	use std::str::FromStr;
 
 	#[test]
-	fn register_collection_encode_encodes_correctly() {
+	fn test_serialized_length() {
```

Comment:
case already tested in `test_serialized_default` I think this can be removed

Reply:
removed